src/mpid/ch4/netmod/ofi/ofi_win.h

```diff
@@ -14,6 +14,123 @@
+        case MPIDI_CH4I_ACCU_MINLOC_SHIFT:
+            MPIDI_OFI_MPI_ACCU_OP_INDEX(MPI_MINLOC, op_index);
+            break;
+        case MPIDI_CH4I_ACCU_BAND_SHIFT:
+            MPIDI_OFI_MPI_ACCU_OP_INDEX(MPI_BAND, op_index);
+            break;
+        case MPIDI_CH4I_ACCU_BOR_SHIFT:
+            MPIDI_OFI_MPI_ACCU_OP_INDEX(MPI_BOR, op_index);
+            break;
+        case MPIDI_CH4I_ACCU_BXOR_SHIFT:
+            MPIDI_OFI_MPI_ACCU_OP_INDEX(MPI_BXOR, op_index);
+            break;
+        case MPIDI_CH4I_ACCU_LAND_SHIFT:
+            MPIDI_OFI_MPI_ACCU_OP_INDEX(MPI_LAND, op_index);
+            break;
+        case MPIDI_CH4I_ACCU_LOR_SHIFT:
+            MPIDI_OFI_MPI_ACCU_OP_INDEX(MPI_LOR, op_index);
+            break;
+        case MPIDI_CH4I_ACCU_LXOR_SHIFT:
+            MPIDI_OFI_MPI_ACCU_OP_INDEX(MPI_LXOR, op_index);
+            break;
+        case MPIDI_CH4I_ACCU_REPLACE_SHIFT:
+            MPIDI_OFI_MPI_ACCU_OP_INDEX(MPI_REPLACE, op_index);
+            break;
+        case MPIDI_CH4I_ACCU_NO_OP_SHIFT:
+            MPIDI_OFI_MPI_ACCU_OP_INDEX(MPI_NO_OP, op_index);
+            break;
+        case MPIDI_CH4I_ACCU_CSWAP_SHIFT:
+            MPIDI_OFI_MPI_ACCU_OP_INDEX(MPI_OP_NULL, op_index);
+            break;
+        default:
+            MPIR_Assert(hint_shift < MPIDI_CH4I_ACCU_OP_SHIFT_LAST);
+            break;
+    }
+    return op_index;
+}
+
+#undef FUNCNAME
+#define FUNCNAME MPIDI_OFI_load_acc_hint
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+MPL_STATIC_INLINE_PREFIX void MPIDI_OFI_load_acc_hint(MPIR_Win * win)
+{
+    int op_index = 0, i;
+    MPIDI_CH4U_win_info_accu_op_shift_t hint_shift = 0;
+
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_OFI_WIN_LOAD_ATOMIC_INFO);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_OFI_WIN_LOAD_ATOMIC_INFO);
+
+    if (!MPIDI_OFI_WIN(win).acc_hint)
+        MPIDI_OFI_WIN(win).acc_hint = MPL_malloc(sizeof(MPIDI_OFI_win_acc_hint_t), MPL_MEM_RMA);
+
+    /* TODO: we assume all pes pass the same hint. Allreduce is needed to check
+     * the maximal value or the spec must define it as same value on all pes.
+     * Now we assume the spec requries same value on all pes. */
+
+    /* We translate the atomic op hints to max count allowed for all possible atomics with each
+     * datatype. We do not need more specific info (e.g., <datatype, op>, because any process may use
+     * the op with accumulate or get_accumulate.*/
+    for (i = 0; i < MPIDI_OFI_DT_SIZES; i++) {
+        MPIDI_OFI_WIN(win).acc_hint->dtypes_max_count[i] = 0;
+        unsigned first_valid_op = 1;
+
+        for (hint_shift = 0; hint_shift < MPIDI_CH4I_ACCU_OP_SHIFT_LAST; hint_shift++) {
+            uint64_t max_count = 0;
+            /* Calculate the max count of all possible atomics if this op is enabled.
+             * If the op is disabled for the datatype, the max counts are set to 0 (see util.c).*/
+            if (MPIDI_CH4U_WIN(win, info_args).which_accumulate_ops & (1 << hint_shift)) {
```

Comment:
What happens if `which_accumulate_ops == 0`? Does it leave `dtypes_max_count[i]` uninitialized?

Reply:
`dtypes_max_count[i]` is always first initialized at line 97.
```
    for (i = 0; i < MPIDI_OFI_DT_SIZES; i++) {
        MPIDI_OFI_WIN(win).acc_hint->dtypes_max_count[i] = 0;
...
```